### PR TITLE
Adam/delete tenant

### PR DESCRIPTION
**Pull request by harageth**
# Resolves

https://jira.rax.io/browse/SALUS-956

# What

Delete all monitors for a particular tenant

# How

Utilizes the new repository function to delete all monitors associated with a tenant.

## How to test

Unit tests have been added

# Why

It seems silly to not do this and instead delete everything individually.

### CODE REVIEW

File: src/main/java/com/rackspace/salus/monitor_management/web/controller/MonitorApiController.java [hunk 1]
```diff
@@ -344,4 +344,10 @@ public void delete(@PathVariable String tenantId,
     return PagedContent.fromPage(monitorManagement.getMonitorsBySearchString(tenantId, searchCriteria, page)
       .map(monitor -> monitorConversionService.convertToOutput(monitor)));
   }
+
+  @DeleteMapping("/admin/tenant/{tenantId}")
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/monitor_management/web/controller/MonitorApiController.java:348]: It looks like this URL path will clash with the zones one. Probably needs to be
```suggestion
  @DeleteMapping("/admin/tenant/{tenantId}/monitors")
```

File: src/main/java/com/rackspace/salus/monitor_management/web/controller/ZoneApiController.java [hunk 2]
```diff
@@ -251,4 +252,10 @@ public void delete(HttpServletRequest request) {
         zoneManagement.getMonitorsForZone(tenantId, zone, pageable)
         .map(MonitorDTO::new));
   }
+
+  @DeleteMapping("/admin/tenant/{tenantId}")
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/monitor_management/web/controller/ZoneApiController.java:256]: and
```suggestion
  @DeleteMapping("/admin/tenant/{tenantId}/zones")
```

File: src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java [hunk 3]
```diff
@@ -319,4 +319,8 @@ public boolean exists(String tenantId, String zoneName) {
     public boolean publicZoneExists(String zoneName) {
         return exists(ResolvedZone.PUBLIC, zoneName);
     }
+
+    public void removeAllTenantZones(String tenantId) {
+      zoneRepository.deleteAllByTenantId(tenantId);
+    }
```
[the remark this answers is not in the document]
Comment (harageth) [reply src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java:325]: I thought I had done that... But maybe it was the envoy tokens.

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java [hunk 4]
```diff
@@ -2252,4 +2252,8 @@ private String getEnvoyIdForResource(Resource resource) {
   public Page<Monitor> getMonitorsBySearchString(String tenantId, String searchCriteria, Pageable page) {
     return monitorRepository.search(tenantId, searchCriteria, page);
   }
+
+  public void removeAllTenantMonitors(String tenantId) {
+    monitorRepository.deleteAllByTenantId(tenantId);
+  }
```
[the remark this answers is not in the document]
Comment (harageth) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java:2258]: Oh right I still need to add this.

File: src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java [hunk 5]
```diff
@@ -320,7 +321,13 @@ public boolean publicZoneExists(String zoneName) {
...
 
-    public void removeAllTenantZones(String tenantId) {
-      zoneRepository.deleteAllByTenantId(tenantId);
+    public void removeAllTenantZones(String tenantId, boolean sendEvents) {
+      if(sendEvents) {
+        Page<Zone> zones = zoneRepository.findAllByTenantId(tenantId, Pageable.unpaged());
+
+        zones.forEach(zone -> removeZone(zone));
+      }else {
+        zoneRepository.deleteAllByTenantId(tenantId);
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java:330]: Still needs removals of public zones from etcd?

Comment (harageth) [reply src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java:330]: https://github.com/racker/salus-telemetry-monitor-management/blob/master/src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java#L198-L209

It isn't happening in the first place. Perhaps I should still check to make sure there aren't any envoys connected?

Comment (itzg) [reply src/main/java/com/rackspace/salus/monitor_management/services/ZoneManagement.java:330]: Thanks for providing that link. I guess the etcd aspect is fully driven by Envoy presence or not (and probably zone-watcher). I think it's fine to skip the Envoy check.

**Pull request merged**